core/src/main/scala/cats/MonadDefer.scala

```diff
@@ -0,0 +1,21 @@
+package cats
+
+import simulacrum.typeclass
+import cats.Eval.always
+
+/**
+ * A [[Monad monad]] that allows for arbitrarily delaying the
+ * evaluation of an operation, triggering its execution on each run.
+ *
+ * @see [[ApplicativeEval]] for capturing effects in an `F[_]`
+ *     applicative context, but without the repeating side-effects
+ *     requirement.
+ */
+@typeclass trait MonadDefer[F[_]] extends Monad[F] with ApplicativeEval[F] {
```

Comment:
So I've added laws that `eval` should indeed be equivalent with `pure(()).map(_ => a)` and `defer` should be equivalent with `pure(()).flatMap(_ => fa)`.

However `Id` cannot implement `MonadDefer`, because for this type-class we are specifying that any *side effects* should be *repeated on each evaluation*, with `defer` building a factory of `F[A]`.

This means that `MonadDefer` can be implemented for `cats.Eval`, `Task`, `IO`, `Coeval`, `Observable` or other monads that handle side-effects, but cannot be implemented for `Id`.  It would fail the specified laws - tried to come up with something meaningful there, would appreciate feedback on those.

`Id` can implement `ApplicativeEval`, which doesn't have such a restriction. `Try` and `Future` are other candidates for `ApplicativeEval` and that can't implement `MonadDefer`. I find `MonadDefer` cool because it describes laziness.